server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/utils/quotas/QuotaLoader.scala

```diff
@@ -0,0 +1,29 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.utils.quotas
+
+import org.apache.james.jmap.mail.Quotas
+import org.apache.james.mailbox.exception.MailboxException
+import org.apache.james.mailbox.model.MailboxPath
+
+trait QuotaLoader {
+  @throws[MailboxException]
+  def getQuotas(mailboxPath: MailboxPath): Quotas
```

Comment:
In fact, loading quota (even in the pre-loaded case) can trigger synchronous DB reads. I propose we rather return a SMono here.

Reply:
Hmmm we use getQuotas as well in the mailboxFactory to populate the quotas when building the mailbox... Not sure if I should have then a reactive method and a non reactive one, or using a block (which I don't think is right), ... any insight?

Reply:
The factory should then return a `SMono[Mailbox]`